src/wallet/components/WalletDropdownBaseName.tsx

```diff
@@ -0,0 +1,63 @@
+import type { Address } from 'viem';
+import { base } from 'viem/chains';
+import { useAccount } from 'wagmi';
+import { useName } from '../../identity/hooks/useName';
+import { Spinner } from '../../internal/components/Spinner';
+import { baseNameSvg } from '../../internal/svg/baseNameSvg';
+import { cn, pressable, text } from '../../styles/theme';
+import type { WalletDropdownBaseNameReact } from '../types';
+
+export function WalletDropdownBaseName({
+  className,
+}: WalletDropdownBaseNameReact) {
+  const account = useAccount();
+
+  const { data: baseName, isLoading } = useName({
+    address: account.address as Address,
```

Comment:
why are you force casting this?

Reply:
If I don't force cast then I get the error:

```
Type '`0x${string}` | undefined' is not assignable to type '`0x${string}`'.
  Type 'undefined' is not assignable to type '`0x${string}`'.ts(2322)
types.ts(186, 3): The expected type comes from property 'address' which is declared here on type 'UseNameOptions'
```

Reply:
Still force casting, but updated to:

```
  const { address } = useAccount();

  const { data: baseName, isLoading } = useName({
    address: address as Address,
    chain: base,
  });
```

Reply:
To avoid type casting we can do:

```
  const { address } = useAccount();

  if (!address) {
    throw new Error(
      'Name: an Ethereum address must be provided to the Identity or Name component.',
    );
  }

  const { data: baseName, isLoading } = useName({
    address: address,
    chain: base,
  });
```

Reply:
Updated to:

```
  const { address } = useAccount();

  if (!address) {
    throw new Error(
      'WalletDropdownBaseName: Address is required.',
    );
  }

  const { data: baseName, isLoading } = useName({
    address: address,
    chain: base,
  });
```